platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -368,4 +394,90 @@ private int findJoystickDevice(int device_id) {
 
 		return -1;
 	}
+
+	/**
+	 *
+	 * @param event
+	 * @param isCaptured - must be used and not MOUSE_MODE_CAPTURED because requestPointerCapture is not atomic and can make a race condition on fast mouse movement right after request
+	 */
+	private boolean handleMouseEvent(MotionEvent event, final boolean isCaptured) {
+		if (event.getActionMasked() == MotionEvent.ACTION_HOVER_ENTER ||
+				event.getActionMasked() == MotionEvent.ACTION_HOVER_MOVE ||
+				event.getActionMasked() == MotionEvent.ACTION_HOVER_EXIT) {
+			final int x = Math.round(event.getX());
+			final int y = Math.round(event.getY());
+			final int type = event.getAction();
+			queueEvent(new Runnable() {
+				@Override
+				public void run() {
+					GodotLib.hover(type, x, y);
```

Comment:
As mentioned above adds the source as well. This will simplify the processing in the native code.